interface/modules/custom_modules/oe-module-weno/src/Services/PharmacyService.php

```diff
@@ -0,0 +1,229 @@
+                $data['primary_pharmacy'],
+                $data['alternate_pharmacy'],
+            ]);
+        } catch (Exception $e) {
+            return $e->getMessage();
+        }
+    }
+
+    function updatePatientWenoPharmacy($pid, $data){
+        //check if pharmacies already exist for patient
+        if(!$this->getWenoPharmacy($pid)){
+            return $this->createWenoPharmaciesForPatient($pid,$data);
+        }
+        $sql = "UPDATE weno_assigned_pharmacy SET "
+        . "primary_ncpdp = ?, "
+        . "alternate_ncpdp = ? "
+        . "WHERE pid = ?";
+
+        try {
+            sqlInsert($sql, [
+                $data['primary_pharmacy'],
+                $data['alternate_pharmacy'],
+                $pid
+            ]);
+        } catch (Exception $e) {
+            return $e->getMessage();
+        }
+    }
+
+    function getPatientPrimaryPharmacy($pid)
+    {
+        $sql = "SELECT pd.pharmacy_id, pd.pid, p.id, p.name, " .
+            "a.foreign_id, a.city, a.line1 FROM patient_data pd " .
+            "LEFT JOIN pharmacies p ON pd.pharmacy_id = p.id " .
+            "LEFT JOIN addresses a ON p.id = a.foreign_id WHERE pd.pid = ?";
+        $result = sqlQuery($sql, [$pid]);
+        return json_encode($result);
+    }
+
+    public function getWenoPharmacyForPatient($pid)
+    {
+        if($pid !== 0){
+            $sql = "SELECT p.pid,p.weno_pharmacy, w.state,w.ncpdp,w.business_name," .
+            "w.address_line_1 FROM patient_data p " .
+            "INNER JOIN weno_pharmacy w ON p.weno_pharmacy = w.ncpdp WHERE pid = ?";
+            $result = sqlQuery($sql,array($pid));
+        }
+        $pharmacy_data = array(
+            "name"  => $result['business_name'] ?? '',
+            "ncpdp"     => $result['ncpdp']  ?? '',
+            "state"     => $result['state']  ?? '',
+            "address"     => $result['address_line_1']  ?? ''
+        );
+
+        return json_encode($pharmacy_data);
+    }
+
+    public function getWenoPrimaryPharm($pid){
+        $sql = "SELECT wap.pid, wap.primary_ncpdp, wp.business_name, " .
+            "wp.city, wp.address_line_1, wp.ncpdp, wp.state FROM weno_assigned_pharmacy wap " .
+            "INNER JOIN weno_pharmacy wp ON wap.primary_ncpdp = wp.ncpdp " .
+            "WHERE wap.pid = ?";
+        $result = sqlQuery($sql, array($pid));
+
+        return $result;
+    }
+
+    public function getWenoAlternatePharm($pid){
+        $sql = "SELECT wap.pid, wap.alternate_ncpdp, wp.business_name, " .
+            "wp.city, wp.address_line_1, wp.ncpdp, wp.state FROM weno_assigned_pharmacy wap " .
+            "INNER JOIN weno_pharmacy wp ON wap.alternate_ncpdp = wp.ncpdp " .
+            "WHERE wap.pid = ?";
+        $result = sqlQuery($sql, array($pid));
+
+        return $result;
+    }
+    
+    public function checkWenoPharmacyLog(){
+        $db_exist = sqlStatement("SELECT * FROM weno_download_log LIMIT 1");
+        if(empty($db_exist)){
+            return "empty";
```

Comment:
Why return a string "empty" here when you're returning a boolean if the check is not empty? You could just return the result of `empty()`. Either way, pick a string or a boolean.